src/services/acspDataService.ts

```diff
@@ -29,4 +27,28 @@ export class AcspDataService {
             return Promise.reject(error);
         }
     }
+
+    typeOfBusinessChange (session:Session, acspData: AcspData, selectedOption: string): AcspData {
+        if (acspData.typeOfBusiness !== selectedOption) {
+            session.deleteExtraData(USER_DATA);
+            const clearedAcspData: AcspData = {
```

Comment:
Could we not just have something like this 

acspData = {
                    typeOfBusiness: selectedOption
                };

Reply:
No because acspData also includes "acsp_data_submission" and "links" objects. If they are deleted the put requests to ascp-api fail

Reply:
Gotcha